### PR TITLE
add email retrieve

### PR DESCRIPTION
**Pull request by kyleroche**
add imap fields and test with

 ```
ToolkitStep(
        "get emails from my inbox with a subject containing the word test",
        tool_names=[email_client.name]
    )
```

### CODE REVIEW

File: griptape/tools/email_client/tool.py [hunk 1]
```diff
@@ -10,11 +11,64 @@
...
-    password: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_PASSWORD"})
+    smtp_host: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_HOST"})
+    smtp_port: Optional[int] = field(default=None, kw_only=True, metadata={"env": "SMTP_PORT"})
+    smtp_password: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_PASSWORD"})
     from_email: Optional[str] = field(default=None, kw_only=True, metadata={"env": "FROM_EMAIL"})
-    use_ssl: bool = field(default=True, kw_only=True, metadata={"env": "SMTP_USE_SSL"})
+    smtp_use_ssl: bool = field(default=True, kw_only=True, metadata={"env": "SMTP_USE_SSL"})
+
+    imap_url: Optional[str] = field(default=None, kw_only=True, metadata={"env": "IMAP_URL"})
+    imap_user: Optional[str] = field(default=None, kw_only=True, metadata={"env": "IMAP_USER"})
```
Comment (vasinov) [inline griptape/tools/email_client/tool.py:21]: `imap_email` for consistency?

**Review by vasinov — CHANGES_REQUESTED** [review]
I think it looks good overall. One issue that will come up with real mailboxes is that this action will return too many emails, which will break the token limit. The right way to solve this is to integrate embeddings support but for now we could perhaps add a `max_retrieve_messages` parameter with a reasonable default?

File: griptape/tools/email_client/tool.py [hunk 2]
```diff
@@ -10,11 +11,64 @@
 
 @define
 class EmailClient(BaseTool):
-    host: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_HOST"})
-    port: Optional[int] = field(default=None, kw_only=True, metadata={"env": "SMTP_PORT"})
-    password: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_PASSWORD"})
+    smtp_host: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_HOST"})
+    smtp_port: Optional[int] = field(default=None, kw_only=True, metadata={"env": "SMTP_PORT"})
+    smtp_password: Optional[str] = field(default=None, kw_only=True, metadata={"env": "SMTP_PASSWORD"})
     from_email: Optional[str] = field(default=None, kw_only=True, metadata={"env": "FROM_EMAIL"})
```
[the remark this answers is not in the document]
Comment (kyleroche) [reply griptape/tools/email_client/tool.py:17]: added username and changed to from email. technically could be different values even if rare

File: griptape/tools/email_client/tool.py (hunk 1 above)
Comment (kyleroche) [reply griptape/tools/email_client/tool.py:21]: could be different from email even if rare

**Comment by kyleroche** [comment]
rearview

**Comment by vasinov** [comment]
Failing tests unrelated to the PR. Merging.

**Pull request merged**